Fabric.Authorization.Client/AuthorizationClient.cs

```diff
@@ -15,13 +18,240 @@ public AuthorizationClient(HttpClient client)
             this.client = client;
         }
 
+        #region Users
+
+        public async Task<UserApiModel> AddUser(UserApiModel userModel)
+        {
+            var message = new HttpRequestMessage(HttpMethod.Post, new UserRouteBuilder().Route)
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(userModel), Encoding.UTF8, "application/json")
+            };
+
+            return await SendAndParseJson<UserApiModel>(message).ConfigureAwait(false);
+        }
+
         public async Task<UserApiModel> GetPermissionsForCurrentUser()
         {
-            var route = AuthorizationRoutes.GetUserPermissionUrl();
+            var route = new UserRouteBuilder().UserPermissionsRoute;
             var message = new HttpRequestMessage(HttpMethod.Get, route);
             return await SendAndParseJson<UserApiModel>(message).ConfigureAwait(false);
         }
 
+        public async Task<List<PermissionApiModel>> GetUserPermissions(string identityProvider, string subjectId)
+        {
+            var message = new HttpRequestMessage(HttpMethod.Get, new UserRouteBuilder().IdentityProvider(identityProvider).SubjectId(subjectId).UserPermissionsRoute);
+            return await SendAndParseJson<List<PermissionApiModel>>(message).ConfigureAwait(false);
+        }
+
+        public async Task AddPermissionsToUser(string identityProvider, string subjectId, List<PermissionApiModel> permissionModels)
+        {
+            var message = new HttpRequestMessage(HttpMethod.Post, new UserRouteBuilder().IdentityProvider(identityProvider).SubjectId(subjectId).UserPermissionsRoute)
+            {
+                Content = new StringContent(JsonConvert.SerializeObject(permissionModels), Encoding.UTF8, "application/json")
+            };
+
+            await SendRequest(message).ConfigureAwait(false);

```

Comment:
Should we consider changing the API to return a user object here? Seems strange that we return a no content on this one...

Reply:
Yeah - another PR definitely makes sense.